changelog/fix-4511-change-registration-domain-docs-link

```diff
@@ -0,0 +1,4 @@
+Significance: minor
+Type: fix
+
+Change Learn More link for the error notification when Apple Pay domain registration failed
```

Comment:
We can make it a bit more readable:

Point the "Learn more" link to a more appropriate document in the Apple Pay domain registration failure notification.